hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -225,6 +129,16 @@ public void addActionDefinition(String tenantId, ActionDefinition actionDefiniti
         if (isEmpty(actionDefinition.getProperties())) {
             throw new IllegalArgumentException("Properties must be not null");
         }
+        String plugin = actionDefinition.getActionPlugin();
+        if (!getActionPlugins().contains(plugin)) {
+            throw new IllegalArgumentException("Plugin: " + plugin + " is not deployed");
+        }
+        Set<String> pluginProperties = getActionPlugin(plugin);
+        actionDefinition.getProperties().keySet().stream().forEach(property -> {
+            if (!pluginProperties.contains(property)) {
+                throw new IllegalArgumentException("Property: " + property + " is not valid on plugin: " + plugin);
+            }
```

Comment:
Right, I understand this check and am OK with an exception.  I was pointing out that the check does not ensure that all [necessary or required] properties are supplied, only that the supplied properties are supported.  It's not clear to me whether an action can pass validation yet fail execution because it did not supply sufficient properties.